lib/Serialization/ModuleFormat.h

```diff
@@ -861,6 +861,7 @@ namespace control_block {
     MODULE_NAME,
     TARGET,
     SDK_NAME,
+    SDK_VERSION,
```

Comment:
@xymus I was trying to use a `main` compiler to compile a `release/6.0` compiler, and I think this change might trigger a crash when the `release/6.0` compiler tries to read some serialization done by the `main` compiler.

If I understand this code correctly, the values in this enumeration are serialized into `.swiftmodule` files. By adding `SDK_VERSION` in the middle of other enumeration cases, it has just taken the value of `REVISION` (as well as offseting any other later cases). If a `main` compiler-generated `.swiftmodule` is tried to be read by a `release/6.0` compiler the values will not match and might try to interpret some blob as the incorrect value.

I can understand that backwards compatibility is not guaranteed, but `validateControlBlock` has a `default:` case that states that it can be the fallback for future versions of the module format, and it would be easy to recover the compatibility with `release/6.0` by moving this towards the end of the enumeration.

Notice that `CHANNEL` (done in #72001) was also added in the middle of the enumeration, and it is in `release/6.0`, but not in `release/5.9`. Moving `CHANNEL` towards the bottom of the enum is also recommended, and that piece should be cherry-picked into `release/6.0` branches.

Reply:
Thanks for pointing this out. I'll move both new fields to the end of the enum on their respective branches.

I'll also add a guard to give up early if the format version doesn't match in `validateControlBlock`. This logic is already in place for mismatches on the `REVISION` field to handle branching issues in release compilers, it would help with these kinds of mistakes as well.